applications/IgaApplication/custom_elements/laplacian_IGA_element.cpp

```diff
@@ -0,0 +1,231 @@
+    NodesArrayType const& ThisNodes,
+    PropertiesType::Pointer pProperties) const
+{
+    return Kratos::make_intrusive<LaplacianIGAElement>(NewId, GetGeometry().Create(ThisNodes), pProperties);
+}
+
+Element::Pointer LaplacianIGAElement::Create(
+    IndexType NewId,
+    GeometryType::Pointer pGeom,
+    PropertiesType::Pointer pProperties) const
+{
+    return Kratos::make_intrusive<LaplacianIGAElement>(NewId, pGeom, pProperties);
+}
+
+// Deconstructor
+
+LaplacianIGAElement::~LaplacianIGAElement()
+{
+}
+
+
+// From classical Laplacian
+void LaplacianIGAElement::CalculateLocalSystem(MatrixType& rLeftHandSideMatrix,
+                                            VectorType& rRightHandSideVector,
+                                            const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    const ProcessInfo& r_process_info = rCurrentProcessInfo;
+    ConvectionDiffusionSettings::Pointer p_settings = r_process_info[CONVECTION_DIFFUSION_SETTINGS];
+    auto& r_settings = *p_settings;
+
+    const Variable<double>& r_unknown_var = r_settings.GetUnknownVariable(); // Temperature
+    const Variable<double>& r_diffusivity_var = r_settings.GetDiffusionVariable(); // Conductivity
+    const Variable<double>& r_volume_source_var = r_settings.GetVolumeSourceVariable(); // HeatFlux
+
+    // reading integration points and local gradients
+    const auto& r_geometry = GetGeometry();
+    const GeometryType::IntegrationPointsArrayType& integration_points = r_geometry.IntegrationPoints(this->GetIntegrationMethod());
+    const GeometryType::ShapeFunctionsGradientsType& DN_De = r_geometry.ShapeFunctionsLocalGradients(this->GetIntegrationMethod());
+    const Matrix& N_gausspoint = r_geometry.ShapeFunctionsValues(this->GetIntegrationMethod());
+
+    const unsigned int number_of_points = r_geometry.size();
+    const unsigned int dim = DN_De[0].size2();
+    
+    //resizing as needed the LHS
+    if(rLeftHandSideMatrix.size1() != number_of_points)
+        rLeftHandSideMatrix.resize(number_of_points,number_of_points,false);
+    noalias(rLeftHandSideMatrix) = ZeroMatrix(number_of_points,number_of_points); //resetting LHS
+    
+    // resizing as needed the RHS
+    if(rRightHandSideVector.size() != number_of_points)
+        rRightHandSideVector.resize(number_of_points,false);
+    noalias(rRightHandSideVector) = ZeroVector(number_of_points); //resetting RHS
+
+    // Initialize DN_DX
+    Matrix DN_DX(number_of_points,dim);
+    Vector temp(number_of_points);
+
+    const double heat_flux = this->GetValue(r_volume_source_var);
+    const double conductivity = this->GetProperties().GetValue(r_diffusivity_var);
+
+    for(IndexType i_point = 0; i_point < integration_points.size(); ++i_point)
+    {
+        noalias(DN_DX) = DN_De[i_point] ; //prod(DN_De[i_point],InvJ0);
+
+        auto N = row(N_gausspoint,i_point);
+        const double IntToReferenceWeight = integration_points[i_point].Weight(); // * std::abs(DetJ0);
```

Comment:
```suggestion
        const double int_to_reference_weight = integration_points[i_point].Weight(); // * std::abs(DetJ0);
```
Please follow style guide (snake_case for local variables and CamelCase for global ones).